computation_graph/composers.py

```diff
@@ -247,14 +258,22 @@ def _infer_composition_edges(
     )
 
 
-def make_compose(
-    *funcs: _ComposersInputType, key: Optional[Text] = None
+@toolz.curry
+def _make_compose_inner(
+    *funcs: _ComposersInputType, key: Optional[Text], is_future: bool
 ) -> base_types.GraphType:
     return toolz.pipe(
         funcs,
         reversed,
         curried.map(_callable_or_graph_type_to_node_or_graph_type),
         curried.sliding_window(2),
-        curried.mapcat(gamla.star(_infer_composition_edges(key=key))),
+        curried.mapcat(
+            gamla.star(_infer_composition_edges(key=key, is_future=is_future)),
```

Comment:
you can compose `_infer_composition_edges` before this function to avoid the signature duplication